src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -426,6 +426,14 @@ public CopyTestCaseResult copyTestCasesToMeasure(
     List<TestCase> copiedTestCases = new ArrayList<>(sourceTestCases.size());
 
     Measure targetMeasure = measureService.findMeasureById(targetMeasureId);
+
+    if (targetMeasure != null && !targetMeasure.getMeasureMetaData().isDraft()) {
+      sourceTestCases.stream()
```

Comment:
instead of looping through the list again, There is already a "For" loop that iterates over sourceTestCases and create a deepcopy of each testCase and persist them. Do you think we can move this setCreatedBeforeVersioning(false) into that block ?

Reply:
Also it is good to have a simple Assert in one of the existing unit test cases.